pkg/types/ccipocr3/chain_accessor.go

```diff
@@ -0,0 +1,313 @@
+package ccipocr3
+
+import (
+	"context"
+	"sort"
+	"time"
+
+	rmntypes "github.com/smartcontractkit/chainlink-ccip/commit/merkleroot/rmn/types"
+	iplugintypes "github.com/smartcontractkit/chainlink-ccip/internal/plugintypes"
+	"github.com/smartcontractkit/chainlink-ccip/plugintypes"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+)
+
+// AccessorMetadata contains metadata about the chain accessor.
+type AccessorMetadata struct {
+	ChainSelector ChainSelector
+	Contracts     map[string]UnknownAddress
+}
+
+// ChainAccessor for all direct chain access. The accessor is responsible for
+// in addition to direct access to the chain, this interface also translates
+// onchain representations of data to the plugin representation.
+type ChainAccessor interface {
+	AllAccessors
+	SourceAccessor
+	DestinationAccessor
+	RMNAccessor
+}
```

Comment:
Very rough organization of methods. These interfaces may become private or change before everything is finished. Some points I'm considering:

* Source/Destination accessors could be top level objects that the CCIPReader has access to rather than the Generic ChainAccessor. This would prevent misusing src/dest only function calls.
* Are the RMN functions globally accessible? It might belong in the All interface.